package jzs.msg;


enum MsgType {
    MSG_SYS_INIT = 0;
    MSG_SYS_TIMER = 1;
    
    MSG_LOW_HEARTBEAT_REQ = 2;
    MSG_LOW_HEARTBEAT_RSP = 3;

    // TDLINK interface
    MSG_TD_ORDER_REQ = 10;
    MSG_TD_ORDER_RSP = 11;
    MSG_TD_ORDER_REPORT_IND = 12;
    MSG_TD_QUERY_ACCOUNT_REQ = 13;
    MSG_TD_QUERY_ACCOUNT_RSP = 14;
    MSG_TD_QUERY_POSITION_REQ = 15;
    MSG_TD_QUERY_POSITION_RSP = 16;
    MSG_TD_CANCEL_ORDER_REQ = 17;
    MSG_TD_CANCEL_ORDER_RSP = 18;
    MSG_TD_QUERY_TRADE_REQ = 80;
    MSG_TD_QUERY_TRADE_RSP = 81;

    // QMS    
    MSG_QMS_MARKETQUOTE_REQ = 20;
    MSG_QMS_MARKETQUOTE_RSP = 21;
    MSG_QMS_BAR_1M_REQ = 22;
    MSG_QMS_BAR_1M_RSP = 23;
    MSG_QMS_STRATEGY_MARKETQUOTES_REQ = 24;
    MSG_QMS_STRATEGY_MARKETQUOTES_RSP = 25;
    MSG_QMS_STRATEGY_SUBSCRIBE_REQ = 26;
    MSG_QMS_STRATEGY_SUBSCRIBE_RSP = 27;

    // MD interface
    MSG_MD_MARKETDATA_IND = 31;    

    // OMS interface
    MSG_OMS_GOAL_PORTFOLIO_REQ = 40;
    MSG_OMS_GOAL_PORTFOLIO_RSP = 41;
    MSG_OMS_QUERY_PORTFOLIO_REQ = 42;
    MSG_OMS_QUERY_PORTFOLIO_RSP = 43;
    MSG_OMS_SET_PANIC_REQ = 44;
    MSG_OMS_SET_PANIC_RSP = 45;
    MSG_OMS_SYNC_ACCOUNT_REQ = 46;
    MSG_OMS_SYNC_ACCOUNT_RSP = 47;
    MSG_OMS_QUERY_ACCOUNT_REQ = 48;
    MSG_OMS_QUERY_ACCOUNT_RSP = 49;
    MSG_OMS_BASKET_ORDER_REQ = 70;
    MSG_OMS_BASKET_ORDER_RSP = 71;
    MSG_OMS_STOP_PORTFOLIO_REQ = 72;
    MSG_OMS_STOP_PORTFOLIO_RSP = 73;

    // EMS Interface
    MSG_EMS_ORDER_REQ = 50;
    MSG_EMS_ORDER_RSP = 51;
    MSG_EMS_ORDER_REPORT_IND = 52;
    MSG_EMS_CANCEL_ORDER_REQ = 53;
    MSG_EMS_CANCEL_ORDER_RSP = 54;
    
    // MISC Interface
    MSG_MISC_GET_TDTICKET_REQ = 60;
    MSG_MISC_GET_TDTICKET_RSP = 61;
    MSG_MISC_SELF_MESSAGE  = 62;

    // Transaction interface
    MSG_TRANS_TRANSACTION = 91;
    MSG_TRANS_GET_TRANSACTION_REQ = 92;
    MSG_TRANS_GET_TRANSACTION_RSP = 93;

    //Order interface
    MSG_ORDER_ORDER = 94;
    MSG_ORDER_GET_ORDER_REQ = 95;
    MSG_ORDER_GET_ORDER_RSP = 96;
};

message MsgHead{    
    required fixed32 tid = 1;
    required string src = 2;
    required string dst = 3;
    optional bool compressed = 4;
};

message Msg {
    required MsgHead head = 1;
    required bytes body = 2;
};


message SysTimer {
    required uint32 id = 1;
    required uint32 interval = 2;
    required int64 data = 3;
};


message SysInit {

};
